scheduler/utilization_based_host_allocator.go

```diff
@@ -27,7 +27,7 @@ type TaskGroupData struct {
 func UtilizationBasedHostAllocator(ctx context.Context, hostAllocatorData HostAllocatorData) (int, error) {
 	distro := hostAllocatorData.Distro
 	numExistingHosts := len(hostAllocatorData.ExistingHosts)
-	if numExistingHosts >= distro.HostAllocatorSettings.MaximumHosts {
+	if distro.Provider != evergreen.ProviderNameDocker && numExistingHosts >= distro.HostAllocatorSettings.MaximumHosts {
```

Comment:
Does max_containers apply somewhere else?

Reply:
Yes, those are checked https://github.com/evergreen-ci/evergreen/blob/master/scheduler/utilization_based_host_allocator.go#L118

Reply:
isn't that just the way that parent hosts are allocated? and the max_containers is only a per-host feature. 

I think we might still want some kind of distro-max containers to prevent one distro taking over all parents in a pool, and also generally giving us a tool to prevent tasks from ddosing the app tier with load. At least as a possibilty. The numbers can be pretty high by default.

Reply:
The parent hosts have their own max_hosts field which is respected. This change is to have the container distro ignore max_hosts because it's meaningless and not settable from the UI, but you can do some weird workflow with switching an existing distro to docker which can set the field. The distro-level check should be left for a different ticket

Reply:
maybe? I think keeping limits on container distros is important to prevent starvation.

Reply:
I agree, but that's not the point of this ticket

Reply:
if we agree that we want to have some limit on containers, and we're taking it out here only to add it back later, that seems... confusing? does the second ticket exist? do you have an idea of how to do that which wouldn't just be reverting this?

Reply:
I view those as 2 separate things. I feel that the right way to prevent unmanageable load is to set a cap on the number of parent hosts. For the potential problem of a single distro taking up all the space on hosts, I feel that the existing behavior of allocation being mostly FIFO will yield the best total makespan in the majority case. Adding a hard cap per distro will either cause inefficiency (if it's hit) or not make any difference (if it's not a hit)

Reply:
if container hosts were first-come-first-serve and only ran one task, then you'd get something that was workable, but they're first-come-first-serve, and can run multiple tasks (until the queue is empty), so you can easily end up in pathological cases with some queues under-served and the containers on the parents pinned until the queue empties. (Similar to the one we see with the host throttle)

Reply:
Yes I agree with you there, but again a static cap isn't the answer to that. You either have the scenario where you set this # below (# of max parent hosts * hosts per pool) in which case you will always be running under full utilization unless you hit the cap, or you set to >= in which case it does nothing. If we actually wanted to prevent this scenario we'd want to adjust teardown behavior, not planning behavior